### ISSUE TITLE
Clarification for accumulation of gradients norms during training 

### ISSUE THREAD

**Issue by tastyminerals** [rivercold/ACClip-Pytorch#1]
It looks like `emb_grad = torch.cat((emb_grad, j.grad.data.view(-1)))` will create an enormous tensor if sampled at each step. Hence, `args.gradient_accumulation_steps` is used to reduce such overhead. 
Do I understand correctly?
